t/results.t

```diff
@@ -62,6 +62,24 @@ like $@, qr/does_not_exist/, 'right error';
 is_deeply $db->query('select * from results_test where name = ?', 'tx3')
   ->hashes->to_array, [], 'no results';
 
+# Savepoints
+my $id;
+{
+  my $tx = $db->begin;
+  $id
+    = $db->query("insert into results_test (name) values ('sp1') returning id")
+    ->array->[0];
+  $tx->savepoint('s1');
+  eval { $db->query("insert into results_test values (?, 'sp2')", $id); };
+  if ($@ && $db->dbh->state == 23505) {
```

Comment:
May be we need a more useful ``Mojo::Pg::Exception`` module which will offer a more convenient interface for handling errors than the comparison dbh state with the code from the documentation?